### PR TITLE
[S] Checks for incapacitation when updating mobility 

### LINKED ISSUE #43295: Dead guy using PDA to message people (me)
## Round ID:104466

If you discovered this issue from playing tgstation hosted servers: Sybil
[Round ID]: 104466

## Testmerges:
 #42780: 'Automatic deadminning' by ShizCalev at commit aee9417772
#43263: 'Test merge change to see the performance impact of client/mousemove binds' by MrStonedOne at commit 76e673023b

## Reproduction:
i Suicided after Messaging everyone.(im the lawyer) i stayed in in my Body after suicide and could still interact with the PDA and even write Messages, and someone answered. I also was buckled to a Chair, so i could still interact with it.

### PR DESCRIPTION
**Pull request by 4dplanner**
## About The Pull Request

For some reason this was checked for living/canusetopic, but not for human/canusetopic. Tell me if this is deliberate, just looks like an oversight

## Why It's Good For The Game
fixes #43295

## Changelog
:cl:
fix: Humans can no longer use UIs while incapacitated
/:cl:

### CODE REVIEW

**Review by SpaceManiac — DISMISSED** [review]

**Comment by ShizCalev** [comment]
we've come full circle, lmao
https://github.com/tgstation/tgstation/commit/3e7184c9756aa0ccc4fa224ffc05171d4ea2dfc3#diff-1ff8328c73a4a8a73ad2c1fe6d517588R724

**Comment by ShizCalev** [comment]
tbh, should probably just move the MOBILITY_UI flag under here; https://github.com/tgstation/tgstation/blob/65a67058694149c4b3dd604f5054275cd6f08c9e/code/modules/mob/living/living.dm#L1108-L1111

**Review by ShizCalev — CHANGES_REQUESTED** [review]
This is just a band-aid that doesn't really fix #43295, since dead/unconscious/ect mobs will still get the `MOBILITY_PULL` flag added to them. `update_mobility()` has to be corrected instead.

Likewise, `canUseTopic()` isn't the only thing that checks if `MOBILITY_UI` exists.

**Comment by 4dplanner** [comment]
You're right, I've fixed it properly.

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -1089,19 +1089,21 @@
...
 			lying = pick(90, 270)
 	else
-		if(!restrained)
-			mobility_flags |= (MOBILITY_UI | MOBILITY_PULL)
-		else
-			mobility_flags &= ~(MOBILITY_UI | MOBILITY_PULL)
 		mobility_flags |= MOBILITY_STAND
 		lying = 0
+	
+	if(should_be_lying || restrained || incapacitated())
```
Comment (ShizCalev) [inline code/modules/mob/living/living.dm]:
```
	if(buckled)
		if(buckled.buckle_lying != -1)
			should_be_lying = buckled.buckle_lying
			(beds = 90)
```
```
	if(should_be_lying || restrained || incapacitated())
 		mobility_flags &= ~(MOBILITY_UI|MOBILITY_PULL)
```

aka
if buckled to bed, canUseTopic = TRUE

**Review by ShizCalev — APPROVED** [review]

**Comment by ShizCalev** [comment]
testmerged on bagil and sybil

**Comment by checkraisefold** [comment]
speedmerge now

**Pull request merged**